frontend/src/store/middlewares/socket/chat-socket.middleware.ts

```diff
@@ -0,0 +1,44 @@
+import { Middleware } from '@reduxjs/toolkit';
+import { SocketEvent, SocketNamespace } from 'common/enums/enums';
+import {
+  AppDispatch,
+  ChatMessageGetAllItemResponseDto,
+} from 'common/types/types';
+import { io } from 'socket.io-client';
+import { chatsActions } from 'store/actions';
+
+type SocketMiddlewareParams = {
+  dispatch: AppDispatch;
+};
+
+const chatSocketInstance = io(SocketNamespace.CHAT);
+
+const chatSocket: Middleware =
+  ({ dispatch }: SocketMiddlewareParams) =>
+  (next) =>
+  (action): void => {
+    if (chatsActions.joinRoom.match(action)) {
+      chatSocketInstance.emit(SocketEvent.CHAT_JOIN_ROOM, action.payload);
+      chatSocketInstance.on(
+        SocketEvent.CHAT_ADD_MESSAGE,
+        (message: ChatMessageGetAllItemResponseDto): void => {
+          dispatch(chatsActions.addMessage(message));
+          // PR Read Messages should be Merged
+          // dispatch(chatsActions.readMessages(message.chatId));
+        },
+      );
+    }
+
+    if (chatsActions.leaveRoom.match(action)) {
+      chatSocketInstance.emit(SocketEvent.CHAT_LEAVE_ROOM, action.payload);
+      chatSocketInstance.off(SocketEvent.CHAT_ADD_MESSAGE);
```

Comment:
Same here, we don't need to unsubscribe from event, app can have cases when we are not in the room but we need to update something, that's why this logic should be handled by backend
```suggestion
```

Reply:
Same as for `.on`, it's to prevent multiple listeners from the same user to the same event